components/fxa-client/ios/FxAClient/FxAccountDeviceConstellation.swift

```diff
@@ -0,0 +1,192 @@
+                        name: .constellationStateUpdate,
+                        object: nil,
+                        userInfo: ["newState": newState]
+                    )
+                }
+            } catch {
+                FxALog.error("Failure fetching the device list: \(error).")
+                return
+            }
+        }
+    }
+
+    func initDevice(name: String, type: DeviceType, capabilities: [DeviceCapability]) {
+        // This is already wrapped in a `fxaQueue.async`, no need to re-wrap.
+        do {
+            try handleAccountAuthErrors { acct in
+                try acct.initializeDeviceSync(name: name, deviceType: type, supportedCapabilities: capabilities)
+            }
+        } catch {
+            FxALog.error("Failure initializing device: \(error).")
+        }
+    }
+
+    func ensureCapabilities(capabilities: [DeviceCapability]) {
+        // This is already wrapped in a `fxaQueue.async`, no need to re-wrap.
+        do {
+            try handleAccountAuthErrors { acct in
+                try acct.ensureCapabilitiesSync(supportedCapabilities: capabilities)
+            }
+        } catch {
+            FxALog.error("Failure ensuring device capabilities: \(error).")
+        }
+    }
+
+    public func setLocalDeviceName(name: String) {
+        DispatchQueue.global().async {
+            do {
+                try self.handleAccountAuthErrors { acct in
+                    try acct.setDeviceDisplayNameSync(name)
+                }
+                self.refreshState()
+            } catch {
+                FxALog.error("Failure changing the local device name: \(error).")
+            }
+        }
+    }
+
+    public func pollForEvents(completionHandler: @escaping (Result<[DeviceEvent], Error>) -> Void) {
+        DispatchQueue.global().async {
+            do {
+                let events = try self.handleAccountAuthErrors { acct in
+                    try acct.pollDeviceCommandsSync()
+                }
+                DispatchQueue.main.async { completionHandler(.success(events)) }
+            } catch {
+                DispatchQueue.main.async { completionHandler(.failure(error)) }
+            }
+        }
+    }
+
+    public func sendEventToDevice(targetDeviceId: String, e: DeviceEventOutgoing) {
+        DispatchQueue.global().async {
+            do {
+                switch e {
+                case let .sendTab(title, url): do {
+                    try self.handleAccountAuthErrors { acct in
+                        try acct.sendSingleTabSync(targetId: targetDeviceId, title: title, url: url)
+                    }
+                }
+                }
+            } catch {
+                FxALog.error("Error sending event to another device: \(error).")
+            }
+        }
+    }
+
+    public func setDevicePushSubscription(sub: DevicePushSubscription) {
+        // No need to wrap in async, this operation doesn't do any IO or heavy processing.
```

Comment:
I'm surprised, I would have thought it would post the updated device record to the server, but maybe I'm misunderstanding the flow of control here..?